test/unit/instrumentation/http2/http2.test.js

```diff
@@ -0,0 +1,684 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+const assert = require('node:assert')
+const test = require('node:test')
+const DESTINATIONS = require('../../../../lib/config/attribute-filter').DESTINATIONS
+const NAMES = require('../../../../lib/metrics/names')
+const hashes = require('../../../../lib/util/hashes')
+const helper = require('../../../lib/agent_helper')
+const Shim = require('../../../../lib/shim').Shim
+const createHttp2ResponseServer = require('./fixtures/http2')
+const events = require('node:events')
+
+test.beforeEach(async (ctx) => {
+  const { server, baseUrl, responses, host, port } = await createHttp2ResponseServer()
+  ctx.nr = {}
+  ctx.nr.initialize = require('../../../../lib/instrumentation/core/http2')
+  ctx.nr.http2 = require('node:http2')
+  ctx.nr.server = server
+  ctx.nr.baseUrl = baseUrl
+  ctx.nr.responses = responses
+  ctx.nr.host = host
+  ctx.nr.port = port
+  ctx.nr.path = '/path'
+  ctx.nr.protocol = 'http'
+  ctx.nr.method = 'POST'
+})
+
+test.afterEach(async (ctx) => {
+  if (ctx.nr?.server) { // not all tests use the actual server
+    ctx.nr.server.destroy()
+  }
+})
+
+test('built-in http2 module instrumentation', async (t) => {
+  await t.test('should not cause bootstrapping to fail', async (t) => {
+    t.beforeEach(async (ctx) => {
+      ctx.nr.agent = helper.loadMockedAgent()
+    })
+
+    t.afterEach((ctx) => {
+      helper.unloadAgent(ctx.nr.agent)
+    })
+
+    await t.test('when passed an empty module', (t) => {
+      const { agent, initialize } = t.nr
+      assert.doesNotThrow(() => initialize(agent, {}, 'http2', new Shim(agent, 'http2')))
+    })
+  })
+
+  await t.test('with outbound request', async (t) => {
+    t.beforeEach(async (ctx) => {
+      const agent = helper.loadMockedAgent()
```

Comment:
if you use `helper.loadMockedAgent` it will call the http2 instrumentation automatically. the segment trees will differ now and the external is the 2nd child of the trace